### PR TITLE
Pocket Lab Updates

### PR DESCRIPTION
**Pull request by D-Disha**
These updates are  to fix the bug reported here :  https://jira.analog.com/browse/PCFW-1204?filter=-1&jql=assignee%20%3D%20gprasad%20%20AND%20resolution%20%3D%20Unresolved%20order%20by%20updated%20DESC 
This bug was present in data capture view as well hence that is fixed too.
The drop_down list size is increased so as to accommodate 16 channels.

### CODE REVIEW

**Comment by D-Disha** [comment]
Hi @ribdp @SaikiranGudla @jansunil @CMinajigi, are there any comments for this PR?

**Comment by SaikiranGudla** [comment]
Looks good to me.
Suggest changing the commit message to something like the below which is more standard,
pocket_lab: Updated xyz.. or
pocket_lab: Fixed so and so...

File: pocket_lab/pl_gui_views.c [hunk 1]
```diff
@@ -394,13 +394,15 @@ void pl_gui_perform_dmm_read(void)
...
-			if (lv_obj_get_state(pl_gui_dmm_chn_checkbox[cnt]) == LV_STATE_CHECKED) {
+			if (lv_obj_get_state(pl_gui_dmm_chn_checkbox[cnt]) == (LV_STATE_CHECKED |
+					LV_STATE_DISABLED)) {
 				/* get DMM reading for current channel and display it into text area */
 				ret = pl_gui_get_dmm_reading(ibuf, cnt, pl_gui_device_indx);
 				if (ret) {
 					return;
 				}
 				lv_textarea_set_text(pl_gui_dmm_chn_ta[cnt], ibuf);
+
```
Comment (jansunil) [inline pocket_lab/pl_gui_views.c:405]: Extra line can be removed

File: pocket_lab/pl_gui_views.c [hunk 2]
```diff
@@ -394,13 +394,15 @@ void pl_gui_perform_dmm_read(void)
 		read_cntr = 0;
 		for (cnt = 0; cnt < pl_gui_dmm_chn_cnt; cnt++) {
 			ibuf[0] = '\0';
-			if (lv_obj_get_state(pl_gui_dmm_chn_checkbox[cnt]) == LV_STATE_CHECKED) {
+			if (lv_obj_get_state(pl_gui_dmm_chn_checkbox[cnt]) == (LV_STATE_CHECKED |
```
Comment (jansunil) [inline pocket_lab/pl_gui_views.c:397]: What is the reason behind this change? Can we have a short description in the commit message?

File: pocket_lab/pl_gui_views.c [hunk 3]
```diff
@@ -879,12 +885,33 @@ static void pl_gui_dmm_btn_event_cb(lv_event_t *event)
...
+					if (lv_obj_get_state(pl_gui_dmm_chn_checkbox[cnt]) == LV_STATE_CHECKED)
+						lv_obj_clear_state(pl_gui_dmm_chn_checkbox[cnt],
+								   LV_STATE_CHECKED | LV_STATE_DISABLED);
+					else
+						lv_obj_clear_state(pl_gui_dmm_chn_checkbox[cnt], LV_STATE_DISABLED);
+				}
+
 				lv_obj_set_style_bg_color(btn, lv_palette_main(LV_PALETTE_GREEN), LV_PART_MAIN);
+				pl_gui_dmm_is_running = !pl_gui_dmm_is_running;
+
```
Comment (jansunil) [inline pocket_lab/pl_gui_views.c:899]: Extra line can be removed

File: pocket_lab/pl_gui_views.c [hunk 4]
```diff
@@ -879,12 +885,33 @@ static void pl_gui_dmm_btn_event_cb(lv_event_t *event)
 		} else { // Start/Stop label
 			if (pl_gui_dmm_is_running) {
 				lv_label_set_text(label, "Start");
+
+				for (cnt = 0; cnt < pl_gui_dmm_chn_cnt; cnt++) {
+					if (lv_obj_get_state(pl_gui_dmm_chn_checkbox[cnt]) == LV_STATE_CHECKED)
+						lv_obj_clear_state(pl_gui_dmm_chn_checkbox[cnt],
+								   LV_STATE_CHECKED | LV_STATE_DISABLED);
+					else
```
Comment (jansunil) [inline pocket_lab/pl_gui_views.c:893]: I suppose if-else blocks with single liner statements within also need to have brackets around. The rest of the file follows such a structure. Can we add brackets everywhere?

File: pocket_lab/pl_gui_views.c [hunk 5]
```diff
@@ -900,19 +927,20 @@ static void pl_gui_capture_btn_event_cb(lv_event_t *event)
 	uint32_t cnt;
 	lv_event_code_t code = lv_event_get_code(event);
 	lv_obj_t *btn = lv_event_get_target(event);
+	int num_of_enabled_channels = 0;
```
Comment (jansunil) [inline pocket_lab/pl_gui_views.c:930]: Do we really need an int here? Can we manage with a uint8_t ?

File: pocket_lab/pl_gui_views.c [hunk 6]
```diff
@@ -924,30 +952,47 @@ static void pl_gui_capture_btn_event_cb(lv_event_t *event)
...
-					if (lv_obj_get_state(pl_gui_capture_chn_checkbox[cnt]) == LV_STATE_CHECKED) {
+					if (lv_obj_get_state(pl_gui_capture_chn_checkbox[cnt]) ==
+					    (LV_STATE_CHECKED | LV_STATE_DISABLED)) {
+						lv_obj_clear_state(pl_gui_capture_chn_checkbox[cnt],
+								   LV_STATE_CHECKED| LV_STATE_DISABLED);
 						lv_chart_remove_series(pl_gui_capture_chart_ovrly, pl_gui_capture_chn_ser[cnt]);
 						pl_gui_capture_chn_ser[cnt] = NULL;
+					} else {
+						lv_obj_clear_state(pl_gui_capture_chn_checkbox[cnt],LV_STATE_DISABLED);
+
```
Comment (jansunil) [inline pocket_lab/pl_gui_views.c:963]: Extra line can be removed

File: pocket_lab/pl_gui_views.c [hunk 7]
```diff
@@ -924,30 +952,47 @@ static void pl_gui_capture_btn_event_cb(lv_event_t *event)
...
 					}
+
```
Comment (jansunil) [inline pocket_lab/pl_gui_views.c:965]: Extra line can be removed

File: pocket_lab/pl_gui_views.c [hunk 8]
```diff
@@ -924,30 +952,47 @@ static void pl_gui_capture_btn_event_cb(lv_event_t *event)
...
+					pl_gui_capture_chn_ser[cnt] = lv_chart_add_series(pl_gui_capture_chart_ovrly,
+								      lv_palette_main(pl_gui_capture_chn_ser_col[cnt]),
+								      LV_CHART_AXIS_PRIMARY_Y);
+					lv_chart_set_point_count(pl_gui_capture_chart_ovrly, 400);
 				}
+				if (num_of_enabled_channels)
+					pl_gui_capture_is_running = !pl_gui_capture_is_running;
 			}
-			pl_gui_capture_is_running = !pl_gui_capture_is_running;
+
```
Comment (jansunil) [inline pocket_lab/pl_gui_views.c:989]: extra line can be removed

File: pocket_lab/pl_gui_views.c [hunk 9]
```diff
@@ -924,30 +952,47 @@ static void pl_gui_capture_btn_event_cb(lv_event_t *event)
...
 		}
+
```
Comment (jansunil) [inline pocket_lab/pl_gui_views.c:991]: extra line can be removed

File: pocket_lab/pl_gui_views.c [hunk 10]
```diff
@@ -924,30 +952,47 @@ static void pl_gui_capture_btn_event_cb(lv_event_t *event)
...
 	}
 }
 
+
```
Comment (jansunil) [inline pocket_lab/pl_gui_views.c:995]: Extra line can be removed

File: pocket_lab/pl_gui_views.c [hunk 11]
```diff
@@ -1060,7 +1105,7 @@ int32_t pl_gui_create_attributes_view(lv_obj_t *parent,
 				      struct pl_gui_init_param *param)
 {
 	int32_t ret;
-	char dropdown_list[100];
+	char dropdown_list[200];
```
Comment (jansunil) [inline pocket_lab/pl_gui_views.c:1108]: What is the reason behind this change?

Comment (D-Disha) [reply pocket_lab/pl_gui_views.c:1108]: I have mentioned it as comment on the PR. I would move it to the commit message for a better clarity.

Comment (jansunil) [reply pocket_lab/pl_gui_views.c:1108]: My bad. I did not notice this

File: pocket_lab/pl_gui_views.c [hunk 12]
```diff
@@ -879,12 +884,33 @@ static void pl_gui_dmm_btn_event_cb(lv_event_t *event)
...
 				lv_obj_set_style_bg_color(btn, lv_palette_main(LV_PALETTE_GREEN), LV_PART_MAIN);
+				pl_gui_dmm_is_running = !pl_gui_dmm_is_running;
 			} else {
 				lv_label_set_text(label, "Stop");
+				for (cnt = 0; cnt < pl_gui_dmm_chn_cnt; cnt++) {
+					if (lv_obj_get_state(pl_gui_dmm_chn_checkbox[cnt]) == LV_STATE_CHECKED) {
+						num_of_enabled_channels++;
+						lv_obj_add_state(pl_gui_dmm_chn_checkbox[cnt],
+								 LV_STATE_CHECKED | LV_STATE_DISABLED);
+					} else
```
Comment (jansunil) [inline pocket_lab/pl_gui_views.c:906]: Brackets for else is still missing here

File: pocket_lab/pl_gui_views.c [hunk 13]
```diff
@@ -879,12 +884,33 @@ static void pl_gui_dmm_btn_event_cb(lv_event_t *event)
...
+						lv_obj_add_state(pl_gui_dmm_chn_checkbox[cnt], LV_STATE_DISABLED);
+				}
+
 				lv_obj_set_style_bg_color(btn, lv_palette_main(LV_PALETTE_RED), LV_PART_MAIN);
 			}
-			pl_gui_dmm_is_running = !pl_gui_dmm_is_running;
+			if (num_of_enabled_channels)
```
Comment (jansunil) [inline pocket_lab/pl_gui_views.c:912]: Brackets are missing here as well

File: pocket_lab/pl_gui_views.c [hunk 14]
```diff
@@ -924,26 +951,38 @@ static void pl_gui_capture_btn_event_cb(lv_event_t *event)
...
+								 LV_STATE_CHECKED | LV_STATE_DISABLED);
+					} else {
+						lv_obj_add_state(pl_gui_capture_chn_checkbox[cnt], LV_STATE_DISABLED);
 					}
+					pl_gui_capture_chn_ser[cnt] = lv_chart_add_series(pl_gui_capture_chart_ovrly,
+								      lv_palette_main(pl_gui_capture_chn_ser_col[cnt]),
+								      LV_CHART_AXIS_PRIMARY_Y);
+					lv_chart_set_point_count(pl_gui_capture_chart_ovrly, 400);
 				}
+				if (num_of_enabled_channels)
```
Comment (jansunil) [inline pocket_lab/pl_gui_views.c:983]: Brackets are missing

Comment (D-Disha) [reply pocket_lab/pl_gui_views.c:983]: Apologies, have missed on this. Will update

**Review by jansunil — APPROVED** [review]

**Review by SaikiranGudla — APPROVED** [review]
